pkg/asset/machines/clusterapi.go

```diff
@@ -281,19 +281,21 @@ func (c *ClusterAPI) Generate(ctx context.Context, dependencies asset.Parents) e
 			return err
 		}
 
+		storageURI := fmt.Sprintf("https://%s.blob.%s", getStorageAccountName(clusterID.InfraID), session.Environment.StorageEndpointSuffix)
```

Comment:
this doesn't look right. i think it should be from the machine pool or empty if Managed/disabled

reading the code (didn't test), I'm not sure if this variable is used once it is passed in to generate. Perhaps this just needs to be cleaned up (deleted).